src/DataTable/Checkbox.js

```diff
@@ -35,6 +36,7 @@ const Checkbox = ({
       aria-label={name}
       checked={checked}
       onChange={() => null} // prevent uncontrolled checkbox warnings -  we don't need onChange
+      disabled={(name !== 'select-all-rows') && disabled}
```

Comment:
as you suggested there was no need of checking for `select-all-row` so that's removed
and now it will show appropriate cursor `pointer` or `not-allowed`